encodings/zstd/src/serde.rs

```diff
@@ -15,6 +15,8 @@ use crate::{ZstdArray, ZstdEncoding, ZstdVTable};
 pub struct ZstdFrameMetadata {
```

Comment:
I wonder how big this metadata is expected to be?

I missed this before, but seems like ZstdMetadata scales with num frames. I wonder if it's worth storing this in a `Buffer<FrameMetadata>` rather than in the protobuf?

Relatedly, is it worth even storing nvalues for primitive typed frames? We could have `uncompressed_sizes: Buffer<u64>; frame_lens: Buffer<u32>` as two buffer children of this array?

Reply:
Are there any operations that would actually benefit from moving this to a buffer? This should be much smaller than the compressed frames, something like 10 bytes per frame meta and 16kB+ (8k*2B) for e.g. uniform u16s.

How would one conditionally omit a protobuf field for primitive types? I think simplicity is also valuable here.

Reply:
How big are frames roughly? Just wondering if there's any value in these being recursively compressed. Maybe overkill though